highfidelity/hifi
# Parabolic Picking + Pointers

https://highfidelity.fogbugz.com/f/cases/17051/Parabolic-Pointer-Implementation

Also: https://highfidelity.manuscript.com/f/cases/16472/Additional-Information-returned-by-findRayIntersection-is-incorrect

Test plan:
- Verify that teleport works as it does in master
- Run all the laser pointer tests: https://github.com/highfidelity/hifi_tests/tree/master/tests/engine/interaction/pointer/laser
- Run all the parabola pointer tests (not yet merged to hifi_tests): https://github.com/SamGondelman/hifi_tests/tree/bd5eeae68e44b75d381235645c3c752fa50d0ee3/tests/engine/interaction/pointer/parabola
- Run [this](https://gist.githubusercontent.com/SamGondelman/7bbb72ec23d3bcc1a9a2651ca5587d5c/raw/6570f92601072ad99116baaaa850f4bc3bc67cdf/ParabolaPointerTest.js).  You'll have a parabola shooting out of your right hand.  Its pitch and yaw should rotate with your hand.
- Verify that the parabola intersects geometry correctly.  Try on various entity types: primitives, models, web, text.  The end of the beam will move to point out of the surface along the normal.
- Verify that this normal looks correct as you move the beam over things.
- Verify that shadows still render.
- Scale your avatar up and down and verify that the end of the beam and the parabola itself scale with you, so the parabola looks the same relative to you at any size.
- Press "u".  The beam will become orange-ish, semi-transparent, and half as wide.  Press it again, the beam will disappear and a pink sphere will be at the end.  Press it again and you'll see the teleport pad.  Verify that the teleport pad rotates to face you as you move it around you.
